File: src/ros_basics_tutorials/src/talker_listener/talker.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import String

def talker():
    # Create a publisher object
    pub = rospy.Publisher('chatter', String, queue_size=10)

    # Create and initialize a ROS node
    rospy.init_node('talker', anonymous=True)

    # Publish the ROS message
    rate = rospy.Rate(1)

    i = 0
    while not rospy.is_shutdown():
        hello_str = "hello world %s" % i
        rospy.loginfo(hello_str)
        pub.publish(hello_str)
        rate.sleep()
        i += 1

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: src/ros_basics_tutorials/src/talker_listener/talker.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <sstream>

int main(int argc, char **argv){
    // Initiate new ROS node
    ros::init(argc, argv, "talker_node");

    // Create a publisher
    ros::NodeHandle n;
    ros::Publisher chatter_publisher = n.advertise<std_msgs::String>("chatter", 10);
    
    // # messages per second
    ros::Rate loop_rate(1);

    int count = 0;
    while (ros::ok()){
        // create a new string msg
        std_msgs::String msg;

        // string for the data
        std::stringstream ss;
        ss << "hello world " << count;
        // assign string data to ROS message
        msg.data = ss.str();

        // print the content in the terminal
        ROS_INFO("[Talker] I published %s\n", msg.data.c_str());
        
        // publish message
        chatter_publisher.publish(msg);
        
        ros::spinOnce();
        loop_rate.sleep();
        count++;
    }
    return 0;
}
